trailmet/algorithms/quantize/quantize.py

```diff
@@ -0,0 +1,181 @@
+
+import torch
+import torch.nn as nn
+import torch.nn.init as init
+from tqdm import tqdm_notebook
+from ..algorithms import BaseAlgorithm
+
+class BaseQuantization(BaseAlgorithm):
+    """base class for quantization algorithms"""
+    def __init__(self, **kwargs):
+        super(BaseQuantization, self).__init__(**kwargs)
+        pass
+
+    def quantize(self, model, dataloaders, method, **kwargs):
+        pass
+    
+    def round_ste(x: torch.Tensor):
+        """
+        Implement Straight-Through Estimator for rounding operation.
+        """
+        return (x.round() - x).detach() + x
+
+    def get_calib_samples(self, train_loader, num_samples):
+        """
+        Get calibration-set samples for finetuning weights and clipping parameters
+        """
+        calib_data = []
+        for batch in train_loader:
+            calib_data.append(batch[0])
+            if len(calib_data)*batch[0].size(0) >= num_samples:
+                break
+        return torch.cat(calib_data, dim=0)[:num_samples]
+
+# the lp_loss, accuracy and test functions can be shifted to the BaseAlogorithm class in ./algorithms.py
+    
+    def lp_loss(pred, tgt, p=2.0, reduction='none'):
+        """loss function measured in Lp Norm"""
+        if reduction == 'none':
+            return (pred-tgt).abs().pow(p).sum(1).mean()
+        else:
+            return (pred-tgt).abs().pow(p).mean()
+
+    def accuracy(self, output, target, topk=(1,)):
```

Comment:
its a general function, can we move it to algorithms or utils?

Reply:
lp_loss is also a generic function, should it be moved to algorithms along with test and accuracy?

https://github.com/heisenbug237/trailmet/blob/33615aff540d4b3883842eca67b1a56538f591a1/trailmet/algorithms/quantize/quantize.py#L36

Reply:
yes please shift it

Reply:
done in e52f86293114e05d095189e26f3c0dcce71d6235